File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

class Main {
    static boolean[] visited;
    static boolean[][] graph;
    static int N,M;
    static int answer;

    public static void DFS(int idx){
        visited[idx] = true;
        answer++;

        for (int i=1; i <= N; i++)
            if (!visited[i] && graph[idx][i])
                DFS(i);

    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());

        graph = new boolean[ N + 1 ][ N + 1 ];
        visited = new boolean[N + 1];

        int x, y;
        for (int i=0; i < M; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            x = Integer.parseInt(st.nextToken());
            y = Integer.parseInt(st.nextToken());
            graph[x][y] = true;
            graph[y][x] = true;
        }

        DFS(1);

        bw.write(String.valueOf(answer -1));
        br.close();
        bw.close();
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.kt
private lateinit var graph: Array<MutableList<Int>>
private lateinit var visited: BooleanArray
private var answer = 0

private fun DFS(node: Int) {
    visited[node] = true
    answer++

    for (i in graph[node]) {
        if (!visited[i]) {
            DFS(i)
        }
    }
}

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()

    val n = br.readLine().toInt()
    val m = br.readLine().toInt()

    graph = Array( n + 1){ mutableListOf() }
    visited = BooleanArray(n + 1)

    repeat(m){
        val (a, b) = br.readLine().split(" ").map { it.toInt() }
        graph[a].add(b)
        graph[b].add(a)
    }

    DFS(1)

    bw.write("${answer - 1}")

    bw.flush()
    bw.close()
    br.close()
}
